### PR TITLE
PP-5553 Add method to generate challenge JWT

### PR DESCRIPTION
**Pull request by stephencdaly**
Make code to generate DDC JWT more generic so parts can be reused to generate challenge JWT.

Add method to Worldpay3dsFlexJwtService to generate challenge JWT.

The exception classes were just renamed to remove "Ddc" from the name, and the exception messages were modified slightly.

### CODE REVIEW

**Comment by stephencdaly** [comment]
Just noticed I forgot to add in the ReturnURL to the JWT

File: src/test/java/uk/gov/pay/connector/charge/service/Worldpay3dsFlexJwtServiceTest.java [hunk 1]
```diff
@@ -51,12 +56,46 @@ public void shouldCreateCorrectTokenForWorldpay3dsFlexDdc() {
...
+
+        Jws<Claims> jws = Jwts.parser()
+                .setSigningKey(new SecretKeySpec(validCredentials.get("jwt_mac_id").getBytes(), "HmacSHA256"))
+                .parseClaimsJws(token);
+
+        assertThat(jws.getHeader().getAlgorithm(), is("HS256"));
+        assertThat((Map<String, Object>)jws.getHeader(), hasEntry("typ", "JWT"));
+        assertThat(jws.getBody(), hasKey("jti"));
+        assertThat(jws.getBody(), hasKey("iat"));
+        assertThat(jws.getBody(), hasEntry("iss", validCredentials.get("issuer")));
```
Comment (oswaldquek) [inline src/test/java/uk/gov/pay/connector/charge/service/Worldpay3dsFlexJwtServiceTest.java]: Do we not need a "ReturnUrl" too?

File: src/test/java/uk/gov/pay/connector/charge/service/Worldpay3dsFlexJwtServiceTest.java [hunk 2]
```diff
@@ -51,12 +56,46 @@ public void shouldCreateCorrectTokenForWorldpay3dsFlexDdc() {
                 .parseClaimsJws(token);
 
         assertThat(jws.getHeader().getAlgorithm(), is("HS256"));
-        assertThat(jws.getHeader().get("typ"), is("JWT"));
-        assertThat(jws.getBody().get("jti"), is(notNullValue()));
-        assertThat(jws.getBody().get("iat"), is(notNullValue()));
-        assertThat(jws.getBody().get("exp"), is(expectedTokenExpirationTimeEpochSeconds));
-        assertThat(jws.getBody().get("iss"), is(validCredentials.get("issuer")));
-        assertThat(jws.getBody().get("OrgUnitId"), is(validCredentials.get("organisational_unit_id")));
+        assertThat((Map<String, Object>)jws.getHeader(), hasEntry("typ", "JWT"));
```
Comment (oswaldquek) [inline src/test/java/uk/gov/pay/connector/charge/service/Worldpay3dsFlexJwtServiceTest.java:99]: should there not be a test for the "alg" too?

**Review by oswaldquek — COMMENTED** [review]
Looks good generally, just two comments

File: src/test/java/uk/gov/pay/connector/charge/service/Worldpay3dsFlexJwtServiceTest.java (hunk 1 above)
Comment (stephencdaly) [reply src/test/java/uk/gov/pay/connector/charge/service/Worldpay3dsFlexJwtServiceTest.java]: Yeh I noticed i'd missed that out, I'll add it in

File: src/test/java/uk/gov/pay/connector/charge/service/Worldpay3dsFlexJwtServiceTest.java (hunk 2 above)
Comment (stephencdaly) [reply src/test/java/uk/gov/pay/connector/charge/service/Worldpay3dsFlexJwtServiceTest.java:99]: as discussed this is tested in the line above which checks the result of `getAlgorithm()`

File: src/test/java/uk/gov/pay/connector/charge/service/Worldpay3dsFlexJwtServiceTest.java (hunk 1 above)
Comment (stephencdaly) [reply src/test/java/uk/gov/pay/connector/charge/service/Worldpay3dsFlexJwtServiceTest.java]: added in the ReturnUrl now

**Review by oswaldquek — APPROVED** [review]

**Pull request merged**